### ISSUE TITLE
Avoid rewriting Shell::getOptionParser() just to get rid of command/subcommand default options.

### ISSUE THREAD

**Issue by bar** [cakephp/cakephp#2456]
### tl;dr
Is there a way to omit the default options from commands/subcommands in a seamless way / without duplicating code? 

### the problem
Actually, ```Shell::getOptionParser()``` passes [no information](https://github.com/cakephp/cakephp/blob/master/lib/Cake/Console/Shell.php#L466) to ```ConsoleOptionParser``` so that default options can be avoided.

The same happens with subcommands [built from arrays](https://github.com/cakephp/cakephp/blob/master/lib/Cake/Console/ConsoleOptionParser.php#L198).

What should be the best way to avoid setting defaults in commands or subcommands without incurring in duplicating code?

If I were to make a PR targeted to ```2.5``` to enable this behavior, what would you recommend?

Full thread here: https://github.com/cakephp/cakephp/pull/2435

**Comment by lorenzo** [comment]
Closing all enhancement requests that were not actioned in a long time. If you still think this a good idea and would like to help making this happen, please leave a comment.
